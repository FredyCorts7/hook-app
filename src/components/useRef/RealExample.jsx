import React, { useState } from 'react';
import '../useEffect/effect.css';
import MultipleCustomHooks from '../examples/MultipleCustomHooks';

const RealExample = () => {
  const [show, setShow] = useState(false);

  return (
    <div>
      <h1>Real Example</h1>
      <hr />

      {show && <MultipleCustomHooks />}

      <button
        className='btn btn-info mt-3'
        onClick={() => {
          setShow(!show);
        }}
      >
        {show ? 'Hide' : 'Show'}
      </button>
    </div>
  );
};

export default RealExample;
