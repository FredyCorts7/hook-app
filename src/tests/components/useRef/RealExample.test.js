import React from 'react';
import { shallow } from 'enzyme';
import RealExample from '../../../components/useRef/RealExample';
import '@testing-library/jest-dom';

describe('Testing to <RealExample />', () => {
  test('should show succesfully', () => {
    const wrapper = shallow(<RealExample />);

    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('MultipleCustomHooks').exists()).toBeFalsy();
    expect(wrapper.find('button').text()).toBe('Show');
  });

  test('should show <MultipleCustomHooks />', () => {
    const wrapper = shallow(<RealExample />);
    wrapper.find('button').simulate('click');

    expect(wrapper.find('MultipleCustomHooks').exists()).toBeTruthy();
    expect(wrapper.find('button').text()).toBe('Hide');
  });
});
