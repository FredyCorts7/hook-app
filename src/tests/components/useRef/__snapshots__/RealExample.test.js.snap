// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing to <RealExample /> should show succesfully 1`] = `
<div>
  <h1>
    Real Example
  </h1>
  <hr />
  <button
    className="btn btn-info mt-3"
    onClick={[Function]}
  >
    Show
  </button>
</div>
`;
